src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    protected a: number,
+    protected b: number,
+    protected c: number,
```

Comment:
[CODE KNOWLEDGE] - When you declare constructor parameters with an access modifier in TypeScript, it implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to manually assign them inside the constructor body.